### PR TITLE
Add reuse lint check to CI and correct licensing information to files

### PR DESCRIPTION
**Pull request by gromakovsky**
## Description

Problem: we want to keep track of copyright and license information in
the repo to make it easier to avoid various legal issues.

Solution: use `reuse lint` to check that all legal information is
present in the repo. Add a step to the CI config to run this command.
Add copyright and license information to all files in the repo.

## Related issue(s)

https://issues.serokell.io/issue/EDNA-15

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,5 @@
+# SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: CC0-1.0
```
Comment (worm2fed) [inline .gitignore:3]: Why the license here is CC0-1.0 and in `backend/.gitignore`, `frontend/.gitignore` it is AGPL-3.0-or-later?

File: default.nix [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: CC0-1.0
```
Comment (worm2fed) [inline default.nix:3]: Why the license here is CC0-1.0 and in `analysis/default.nix`, `backend/default.nix`, `docker.nix`, `flake.nix`, `frontend/default.nix` it is AGPL-3.0-or-later?

File: shell.nix [hunk 3]
```diff
@@ -1,3 +1,7 @@
+# SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+#
+# SPDX-License-Identifier: CC0-1.0
```
Comment (worm2fed) [inline shell.nix:3]: Why the license here is CC0-1.0 and in `analysis/default.nix`, `backend/default.nix`, `docker.nix`, `flake.nix`, `frontend/default.nix` it is AGPL-3.0-or-later?

**Review by worm2fed — CHANGES_REQUESTED** [review]
I have a few questions about different licenses in similar files. And also there are errors with tsx files. Rest of changes seems ok

**Comment by gromakovsky** [comment]
@worm2fed the general rule is the following: `AGPL-3.0-or-later` applies to all files by default, but some files are likely not copyrightable and the license doesn't matter there, so we set it to `CC0-1.0` to make it more explicit.
https://www.notion.so/serokell/FAQ-8da3477b8b7c4faaab6460ff38fc8929

There is no formal strict rule to distinguish these kinds of files. I put `CC0-1.0` somewhere and maybe I've missed some files. But AFAIU the license doesn't matter in non-copyrightable files anyway, `CC0-1.0` just makes it clearer, but `AGPL-3.0-or-later` shouldn't cause troubles. So even if there are more non-copyrightable files with `AGPL-3.0-or-later``, I suppose it's not a big deal.

All `.gitignore` files now have `CC0-1.0`. Regarding `.nix` files, my logic was the following: there are some files that exist just to make old nix work with flakes, they are small "compatibility shims" copy-pasted everywhere (I mean that each repo that uses flakes with old nix has these files), so they are likely not copyrightable. These files are `default.nix` and `shell.nix` in the root. So they have `CC0-1.0`. Other `.nix` files were written by us from scratch and contain something not entirely trivial, so they have `AGPL-3.0-or-later`.

**Review by worm2fed — APPROVED** [review]
Okay, for now it's clear.

**Review by kirelagin — APPROVED** [review]
There is a new REUSE recommendation: https://reuse.software/faq/#license-templates.

tl;dr they recommend not to bother with `LicenseRef-` for stupid licenses that embed the copyright notice. Instead they suggest to just include the original unmodified text of the license template (i.e. straight with `<copyright holders>`), since, even if the license says to preserve the copyright notice, the idea is that probably modifying it to be an SPDX header and is acceptable.
